app/domain_models/follow_and_unfollow_user.rb

```diff
@@ -0,0 +1,61 @@
+require_relative "follow_status_data"
+class FollowAndUnfollowUser
+
+    attr_reader :subject_user_id, :follow_status_data
+  
+    def initialize(subject_user_account)
+      @subject_user_id=User.find_by(account: subject_user_account).id
+      @follow_status_data=FollowStatusData.new()
+    end
+
+    def follow(object_user_account)
+        object_user_id=User.find_by(account: object_user_account).id
+        cannot_follow_yourself!(object_user_id)
+        already_following!(object_user_id)
+        follow_status_data.add_follow_status_record(subject_user_id,"follow",object_user_id)
+        
+    end
+
+    def unfollow(object_user_account)
+        object_user_id=User.find_by(account: object_user_account).id
+        cannot_follow_yourself!(object_user_id)
+        not_following!(object_user_id)
+        follow_status_data.add_follow_status_record(subject_user_id,"unfollow",object_user_id)
+    
+        
+    end
+
+    def cannot_follow_yourself!(object_user_id)
```

Comment:
I think those methods should be in private method.

Reply:
I see, I will make them private in next PR.